language/move-prover/bytecode-to-boogie/src/boogie_wrapper.rs

```diff
@@ -0,0 +1,285 @@
+    ///    output.bpl(2989,5): inline$LibraAccount_pay_from_sender_with_metadata$0$anon0
+    ///    ...
+    /// ```
+    fn extract_verification_errors(&self, out: &str) -> Vec<BoogieError> {
+        let verification_diag_start = Regex::new(r"(?m)^.*Error BP\d+:(?P<msg>.*)$").unwrap();
+        let verification_diag_cond =
+            Regex::new(r"(?m)^.+\((?P<line>\d+),(?P<col>\d+)\): Related.*$").unwrap();
+        let verification_diag_trace = Regex::new(r"(?m)^Execution trace:$").unwrap();
+        let verification_diag_trace_entry =
+            Regex::new(r"(?m)^    .*\((?P<line>\d+),(?P<col>\d+)\): (?P<msg>.*)$").unwrap();
+        let mut errors = vec![];
+        let mut at: usize = 0;
+        while let Some(cap) = verification_diag_start.captures(&out[at..]) {
+            let msg = cap.name("msg").unwrap().as_str();
+            at += cap.get(0).unwrap().end();
+            if let Some(cap) = verification_diag_cond.captures(&out[at..]) {
+                let line = cap.name("line").unwrap().as_str();
+                let col = cap.name("col").unwrap().as_str();
+                let mut trace = vec![];
+                at += cap.get(0).unwrap().end();
+                if let Some(m) = verification_diag_trace.find(&out[at..]) {
+                    at += m.end();
+                    while let Some(cap) = verification_diag_trace_entry.captures(&out[at..]) {
+                        let line = cap.name("line").unwrap().as_str();
+                        let col = cap.name("col").unwrap().as_str();
+                        let msg = cap.name("msg").unwrap().as_str();
+                        trace.push((make_position(line, col), msg.to_string()));
+                        at += cap.get(0).unwrap().end();
+                        if !out[at..].starts_with("\n  ") && !out[at..].starts_with("\r\n  ") {
+                            // Don't read further if this line does not start with an indent,
+                            // as all trace entries do. Otherwise we would match the trace entry
+                            // for the next error.
+                            break;
+                        }
+                    }
+                }
+                errors.push(BoogieError {
+                    kind: BoogieErrorKind::Verification,
+                    position: make_position(line, col),
+                    message: msg.to_string(),
+                    execution_trace: trace,
+                });
+            }
+        }
+        errors
+    }
+
+    /// Extracts compilation errors. This captures any kind of errors different than the
+    /// verification errors (as seen so far).
+    fn extract_compilation_errors(&self, out: &str) -> Vec<BoogieError> {
+        let mut errors = vec![];
+        let diag_re =
+            Regex::new(r"(?m)^.*\((?P<line>\d+),(?P<col>\d+)\).*(Error:|error:).*$").unwrap();
+        for cap in diag_re.captures_iter(&out) {
```

Comment:
```suggestion
        diag_re.captures_iter(&out).iter().map(|cap| ... )
```

would be a be more Rust-y/concise.

Reply:
Done